### ISSUE TITLE
Allow tweaking underlying ganache's configuration

### ISSUE THREAD

**Issue by spalladino** [OpenZeppelin/openzeppelin-test-environment#31]
We should allow tweaking ganache's configuration when running the tests. We currently only support block gas limit and accounts initial ETH, but we should also cover:

- **EVM**: to test how code will behave in a future fork (this includes upgrading the underlying ganache-core version)
- **Forking**: to support testing upgrades or other actions that would be run on mainnet
- **Errors on RPC response**: to test an app's behaviour upon reverts in a more realistic setting (noVMErrorsOnRPCResponse)
- **Port**: to have a consistent port in case the user wants to attach another process to ganache to monitor the transactions being run

We can either do this by allowing the user to spin up their own ganache and attach to it (instead of using the one provided by test-env) or by adding more options to the configuration.

**Comment by frangio** [comment]
We should also support "unlocked accounts", which allows using accounts without knowing their private key. What I've heard is that they're useful when forking.

**Comment by spalladino** [comment]
Thanks Fran, added to the body of the issue!

**Comment by mds1** [comment]
Hey there, just curious what kind of timeline is expected for this issue? Really looking forward to using the OpenZeppelin Test Environment in my project, but am waiting for a few of these settings to be supported.

**Comment by spalladino** [comment]
Hey @mds1! Unfortunately this is not scheduled for the upcoming weeks, so I cannot give a precise answer. We would certainly welcome a PR with the feature if you are interested, though!

**Comment by mds1** [comment]
Got it, thanks. Probably won't get to it for a few weeks, but would be willing to take a shot at the PR then!

**Comment by abcoathup** [comment]
Request on the community forum for support testing on a forked ganache: https://forum.openzeppelin.com/t/testing-on-ganache-fork/2434

**Comment by frangio** [comment]
@knarz has submitted https://github.com/OpenZeppelin/openzeppelin-test-environment/pull/55 enabling support for `allowUnlimitedContractSize`. Rather than adding ad-hoc suport for specific options I think we should add generic support for forwarding an object with arbitrary options. What does everyone think about this? I don't see this specifically mentioned in the comments above.

More specifically, I am proposing adding a field to our configuration field like:

```js
module.exports = {
  accounts: {
    amount: 10, // Number of unlocked accounts
    ether: 100, // Initial balance of unlocked accounts (in ether)
  },

  node: {
    allowUnlimitedContractSize: true,
  },
};
```

where the object under `node` is forwarded as options to Ganache. I'm proposing `node` in case we introduce support for other nodes in the future (e.g. Geth) and the same option forwarding could take place. Alternatively we could just name it `ganache` for now.

There are some options we would not support, like custom account configuration. We would have to detect those and possibly throw an error or simply ignore them.

**Comment by ylv-io** [comment]
Agree with @frangio on grouping. Accounts are also a part of Ganache/Node grouping because eventually Ethereum client is holding them.

**Comment by frangio** [comment]
I disagree with putting accounts inside the `ganache`/`node` object because there is a "preprocesing" step done by test-env. I would expect everything inside that object to be forwarded without any changes.
